### ISSUE TITLE
Cant remove snapshot on Windows

### ISSUE THREAD

**Issue by chester89** [prometheus/prometheus#6620]
## Proposal
**Use case. Why is this important?**
Need to be able to delete old snapshot files

## Bug Report
**What did you do?**
tried to remove snapshot directory after taking a snapshot via `curl -XPOST http://localhost:9090/api/v1/admin/tsdb/snapshot`. tried with `skip_head` parameter as well, same issue
**What did you expect to see?**
files been deleted

**What did you see instead? Under which circumstances?**
an error saying "the file is being used by prometheus.exe". Had to stop the service to remove the directory

**Environment**

* System information:

Windows Server 2016

* Prometheus version:

2.11.1
console options `--config.file D:\prometheus\config\prometheus.yml --storage.tsdb.path D:\prometheus\data --storage.tsdb.retention.time 30d --log.level info --web.enable-admin-api`

* Prometheus configuration file:
```
global:
  scrape_interval:     15s # By default, scrape targets every 15 seconds.

  external_labels:
    monitor: 'codelab-monitor'

scrape_configs:
  - job_name: 'win-local'

    scrape_interval: 15s

    static_configs:
      - targets: ['localhost:5015']	
        labels:
          os: windows
          type: box
  - job_name: 'win-sql'

    scrape_interval: 5m

    static_configs:
      - targets: ['localhost:6016']
        labels:
          os: windows
          type: database
          appname: sqlserver
          kind: infra
  - job_name: 'own-metrics'

    scrape_interval: 5m

    static_configs:
      - targets: ['localhost:9090']
        labels:
          os: windows
          type: prometheus
          kind: infra
```

**Comment by chester89** [comment]
I updated to 2.15.1 and the issue is still present

**Comment by chester89** [comment]
By the way, the issue is only present if I try to remove it via Windows UI and/or commands like `rmdir` or Powershell's `Remove-Item` - it gives me `access denied` error.
Using `rm -r` command from MinGW is working fine

**Comment by roidelapluie** [comment]
We fixed windows specific issues in 2.15.2. not related to this but you might want to upgrade.

**Comment by roidelapluie** [comment]
We have looked at this into our bug scrub.

Snapshots in prometheus uses hard link and we believe that Windows is refusing to delete the files that are still open by Prometheus, and there is nothing that we can do at the prometheus level to prevent this.
